### ISSUE TITLE
Author names do not localize in reader front end

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#9184]
**Describe the bug**
When author names are present in multiple languages, changing the front end reader interface into other languages does not result in the author name changing to match. The author names only appear to present in the journal's primary locale (or perhaps the submission's).

See: https://forum.pkp.sfu.ca/t/authors-names-do-not-change-when-switching-the-language-in-3-4/79957/6

**To Reproduce**
1. Enter author names into a submission's metadata in multiple languages
2. Publish the article
3. View the article in the journal's front end
4. Change the language (e.g. using the language toggle block)
5. See that the author name does not change to match the selected locale.

**What application are you using?**
OJS 3.4.0-1

**Additional information**
This appears to be caused by https://github.com/NateWr/pkp-lib/commit/8485e5a56acc8e020ad2c74213e2156a5246eee0, which is tagged with the wrong issue number. I believe the correct issue number is https://github.com/pkp/pkp-lib/issues/7477.

<hr>

PRs:

stable-3_4_0:
- https://github.com/pkp/googleScholar/pull/17
- https://github.com/pkp/pkp-lib/pull/9219
- https://github.com/pkp/ojs/pull/4001
- https://github.com/pkp/omp/pull/1438 
- https://github.com/pkp/ops/pull/552 

main:
- https://github.com/pkp/googleScholar/pull/18
- https://github.com/pkp/pkp-lib/pull/9240
- https://github.com/pkp/ojs/pull/4011
- https://github.com/pkp/omp/pull/1446
- https://github.com/pkp/ops/pull/558

**Comment by defstat** [comment]
I think that removing https://github.com/pkp/pkp-lib/blob/main/classes/author/Author.php#L98-L109 solves this (at least) issue. Regarding the `Author` `FullName`, the submission locale has a "priority" against the UI locale in the current code base.

**Comment by bozana** [comment]
@asmecher, could you maybe review the PRs above? I also changed the oai dc:creator to be multilingual. 
And it seems oai marc and marcxml use journal primary locale for the metadata. Do you maybe know if there is a specific reason for it? If not I would change that as well, maybe in another issue, to use the submission primary locale.
Thanks a lot!

**Comment by asmecher** [comment]
@bozana, thanks, those look good! For choice of locale in 3rd-party data feeds, I consider the "bug" to be the lack of support for multilingual metadata in those formats rather than the choice of which to use. I don't think many people are using MARC/MARCXML and am hesitant to change it without specific feedback, so I'd suggest trying not to disturb it unless we have a clear reason to improve it.

**Comment by asmecher** [comment]
@touhidurabir is working on changes that [touch some of the same code](https://github.com/pkp/pkp-lib/issues/8025) you are working with; would you mind chatting with him to see if you are coordinated on approaches?

**Comment by bozana** [comment]
Thanks @asmecher!
I have spoken with @touhidurabir today and his changes do not touch the code I have changed, but we realized that all those getLocalizedData + getLocalePrecedence could/should maybe be implemented only in one place, considering those special cases like Identity and Author, that override the getLocalizedData() and have different locale precedence.
Thus, I will merge these changes and then we can see how to bind the getLocalizedData + getLocalePrecedence at one place.
